cache.c

```diff
@@ -465,14 +468,22 @@ struct track_info *cache_get_ti(const char *filename, int force)
 
 struct track_info **cache_refresh(int *count, int force)
 {
-	struct track_info **tis = get_track_infos();
+	struct track_info **tis = get_track_infos(true);
 	int i, n = total;
+	uint64_t last_break = time_in_ms();
 
 	for (i = 0; i < n; i++) {
 		unsigned int hash;
 		struct track_info *ti = tis[i];
 		struct stat st;
 		int rc = 0;
+		uint64_t now;
+
+		now = time_in_ms();
+		if (now - last_break > 100) {
+			cache_yield();
+			last_break = time_in_ms();
+		}
```

Comment:
Maybe it's better to yield unconditionally here. cache_yield is pretty efficient.